configs/components/_base-ruby.rb

```diff
@@ -31,10 +31,15 @@
 elsif platform.is_solaris?
   # See PA-5639, if we decide to go without OpenCSW GCC then we can simplify this logic
   if ruby_version_y >= '3.0'
-    pkg.environment 'PATH', "#{settings[:bindir]}:/opt/csw/bin:/usr/ccs/bin:/usr/sfw/bin:$(PATH)"
-    pkg.environment 'CC', '/opt/csw/bin/gcc'
-    pkg.environment 'LD', '/opt/csw/bin/gld'
-    pkg.environment 'AR', '/opt/csw/bin/gar'
+    if platform.name == 'solaris-113-sparc'
```

Comment:
For future use, would it be simpler just look for `=~ /sparc/`?

Reply:
Yeah that would work here because we're in the if `ruby >= 3` block, but in other places like augeas and ffi, we're not. So to make it clearer how this block of code is related to the code in the other components, I figured it'd be clearer to call out `113`. At some point down the road, it would great to update the Solaris 11 Intel and SPARC templates to 11.4 (with latest SRU) and standardize on one compiler toolchain and remove all of this special casing.

Reply:
Yeah thinking about this more, I think using `! platform.is_cross_compiled? && platform.architecture == 'sparc'` would communicate the same thing (when performing native compiles on sparc), but not be tied to a specific template.